src/lib/zf/tcp_out.c

```diff
@@ -1032,14 +1033,13 @@ static int tcp_output_segment(struct zf_tcp* tcp, struct tcp_seg *seg)
   /* Stop zwin timer */
   zf_tcp_timers_timer_stop(tcp, ZF_TCP_TIMER_ZWIN);
 
-  /* Do not retransmit a segment which has not got out of NIC. */
```

Comment:
This seems like an oversight in previous commits. The comment has not been valid for quite a while.
Originally no packet was sent in this case so comment was valid. Then change was made to actually send the packet in a 'fire and forget mode'. 

When TX timestamping was added this was potentially overlooked. Worth running the test with this patch in debug with logging enabled to ensure that the pkt buffers are still being free'd and re-used as expected.

Reply:
This is indeed the case, thanks @ibeecraft-amd! I removed this comment because it misled me when I first read it. The new comment should better reflect our actual intention with this bit of code, and brings in the removed comment below that (as we wouldn't want to "forget" about this packet).

Reply:
For completeness, the above comment was:
- added in Xilinx-CNS/tcpdirect_history@da6bdc033fd97ad7cb7e769a427a0e178d534ec6
- invalid as of Xilinx-CNS/tcpdirect_history@2da2de10474981438afa593a9de826c9a40b4a50

And the failure mode observed here has been present since Xilinx-CNS/tcpdirect_history@59d0a4ec80c3d95b756d293a149c7a08e9e4c0ed.